CMakeLists.txt

```diff
@@ -126,11 +126,24 @@ function(clr_unknown_arch)
     message(FATAL_ERROR "Only AMD64, ARM and I386 are supported")
 endfunction()
 
+if(INTL_ICU_SH)
+    unset(INTL_ICU_SH CACHE)    # don't cache
+    add_definitions(-DINTL_ICU=1)
+    set(ICU_INTL_ENABLED 1)
+endif()
+
 if(ICU_INCLUDE_PATH)
     add_definitions(-DHAS_REAL_ICU=1)
     set(ICU_CC_PATH "${ICU_INCLUDE_PATH}/../lib/")
     find_library(ICUUC icuuc PATHS ${ICU_CC_PATH} NO_DEFAULT_PATH)
-    find_library(ICU18 icui18n PATHS ${ICU_CC_PATH} NO_DEFAULT_PATH)
+    if(ICU_INTL_ENABLED)
+        find_library(ICU18 icui18n PATHS ${ICU_CC_PATH} NO_DEFAULT_PATH)
+        # icu header files are either located under the same folder or i18n is under a relative path
+        # TODO (unlikely): shall we add `--icu-intl` to build.sh ?
```

Comment:
Are we making a distinction between --icu-intl (as in the path we want to specify) versus --intl-icu (not present on CMake builds, terminology used on Windows builds i.e. /p:IntlICU=true), versus --with-intl?

I agree this is not likely necessary and can be skipped for now.